### PR TITLE
bfrec: Support standard capsule update

### PR DESCRIPTION
**Pull request by lsun100**
This commit adds support for the standard capsule upgrade. The
same capsule file MmcBootCap will be used as default file but
it contains a real capsule and will be copied to the standard
directory <efi_partition>/EFI/UpdateCapsule/ when applied. For
old version of boot images without this capsule support it'll
fall back to use default bootctl mode.

### CODE REVIEW

File: bfrec [hunk 1]
```diff
@@ -44,8 +44,8 @@ Usage: $PROGNAME    [--help]
 
 Description:
   --help                : print help.
-  --default [FILE]      : update the boot partition via the
-                          kernel path (deprecated). If FILE
+  --default [FILE]      : update the boot partition use default
```
Comment (kblaiech) [inline bfrec:47]: this is previously marked as "depracated". It is intended to be removed in the future. Plus I don't think "default mode" is clear enough.

[the remark this answers is not in the document]
Comment (lsun100) [reply bfrec]: Done. Removed the '--default since it's obsoleted'.

File: bfrec [hunk 2]
```diff
@@ -188,37 +181,22 @@ uefi_capsule_update()
...
     fi
 
-    # Copy the capsule file into the EFI System Partition. Rename the file
-    # to match the expected pathname in UEFI, if needed.
-    $run cp $capsule_file $mount_efi/$capsule_efi
-    $run cp $capsule_location/$updater_efi $mount_efi/$updater_efi
+    # Copy the capsule file into the EFI System Partition.
+    $run mkdir -p $mount_efi/EFI/UpdateCapsule 2>/dev/null
+    $run cp $capsule_file $mount_efi/EFI/UpdateCapsule/$capsule_efi
+    $run printf "\x07\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00" > \
```
[the remark this answers is not in the document]
Comment (lsun100) [reply bfrec:180]: Done.

File: bfrec (hunk 1 above)
Comment (lsun100) [reply bfrec]: Removed.

File: build-bfb [hunk 3]
```diff
@@ -45,8 +44,7 @@ while [ "$1" != -- ]; do
         --uart) shift; uart="$1" ;;
         --sys) shift; sys="$1" ;;
         --root) shift; root=$1 ;;
-        --mmc-cap) shift; mmc_cap=$1 ;;
-        --spi-img) shift; spi_img=$1 ;;
+        --causple) shift; capsule=$1 ;;
```
Comment (kblaiech) [inline build-bfb:47]: typo "capsule"

Comment (lsun100) [reply build-bfb:47]: Done. BTW, Spencer will remove build-bfb from bfscripts in one of the pending pull request.

**Pull request merged**